fiftyone/core/singletons.py

```diff
@@ -24,7 +26,13 @@ def __new__(metacls, *args, **kwargs):
         return cls
 
     def __call__(cls, name=None, _create=True, *args, **kwargs):
-        instance = cls._instances.pop(name, None)
+        instance = None
+        if name:
+            doc = foo.get_db_conn().datasets.find_one(
```

Comment:
Just calling out that this adds another db connection into the process of loading a dataset. Not impactful for direct mongo connections but would add more overhead to API connections.